azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Subnet.java

```diff
@@ -71,6 +93,7 @@
     interface Definition<ParentT> extends
         DefinitionStages.Blank<ParentT>,
         DefinitionStages.WithAddressPrefix<ParentT>,
+        DefinitionStages.WithNetworkSecurityGroup<ParentT>,
         DefinitionStages.WithAttach<ParentT> {
     }
 
```

Comment:
Another reason for me to include optionals only in `WithCreate`/`WithAttach` was - When `Definition` extends only from all required interfaces and then from `WithCreate`, it was bit easy for me to image the original flow, for example:

```java
    /**
     * The entirety of the network interface definition.
     */
    interface Definition extends
            DefinitionStages.Blank,
            DefinitionStages.WithGroup,
            DefinitionStages.WithPrimaryNetwork,
            DefinitionStages.WithPrimaryNetworkSubnet,
            DefinitionStages.WithPrimaryPrivateIp,
            DefinitionStages.WithCreate {
    }
```

Here by looking at the definition  I know where required chain ends, then when I see `DefinitionStages.WithCreate` I switch to that interface to see all optionals

```java
        interface WithCreate extends
                Creatable<NetworkInterface>,
                Resource.DefinitionWithTags<WithCreate>,
                WithPrimaryPublicIpAddress,
                WithNetworkSecurityGroup,
                WithSecondaryIpConfiguration {
...
}
```

Without this separation, I had to take a intersection of the interfaces that `Definition` and `WithCreate` extends  OR arrange the extends in such a way that required appears before `DefinitionStages.WithCreate` and optional after that.

```java
    interface Definition extends
            DefinitionStages.Blank,
            DefinitionStages.WithGroup,
            DefinitionStages.WithPrimaryNetwork,
            DefinitionStages.WithPrimaryNetworkSubnet,
            DefinitionStages.WithPrimaryPrivateIp,
            WithPrimaryPublicIpAddress,
            WithNetworkSecurityGroup,
            WithSecondaryIpConfiguration,
            DefinitionStages.WithCreate, {
    }
```